xilinx/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+pub mod sys;
+pub use sys::*;
+
+/*---- decoder ----*/
+pub mod xlnx_decoder;
+pub use xlnx_decoder::*;
+
+pub mod xlnx_dec_utils;
+pub use xlnx_dec_utils::*;
+
+pub mod xlnx_dec_props;
+pub use xlnx_dec_props::*;
+
+/*---- error ----*/
+pub mod xlnx_error;
+pub use xlnx_error::*;
+
+const XCLBIN_FILENAME: &str = "/opt/xilinx/xcdr/xclbins/transcode.xclbin";
+
+/// initalized the number of devices specified with the default xclbin_name
+///
+/// @decice_count: The number of devices available in the system.
+pub fn xlnx_init_all_devices(device_count: i32) -> Result<(), simple_error::SimpleError> {
+    let mut xclbin_params = Vec::new();
+
+    let fname = std::ffi::CString::new(XCLBIN_FILENAME).expect("CString::new failed");
+    let fname_ptr = fname.into_raw();
```

Comment:
This looks like a memory leak:

> Consumes the CString and transfers ownership of the string to a C caller.
> 
> The pointer which this function returns must be returned to Rust and reconstituted using [CString::from_raw](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw) to be properly deallocated. Specifically, one should not use the standard C free() function to deallocate this string.
> 
> Failure to call [CString::from_raw](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw) will lead to a memory leak.

It should just be:

```suggestion
    let fname_ptr = fname.as_ptr();
```